pages/admin/permit-holders.tsx

```diff
@@ -74,59 +74,65 @@ const PermitHolders: NextPage = () => {
   // This will avoid firing a query for each key the user presses
   const debouncedSearchFilter = useDebounce<string>(searchFilter, 500);
 
-  // Set page number to 0 after every filter or sort change
-  useEffect(() => {
-    setPageNumber(0);
-  }, [permitStatusFilter, userStatusFilter, debouncedSearchFilter, dateRange]);
-
   // GQL Query
-  useQuery<GetPermitHoldersResponse, GetPermitHoldersRequest>(GET_PERMIT_HOLDERS_QUERY, {
-    variables: {
-      filter: {
-        userStatus: userStatusFilter,
-        permitStatus: permitStatusFilter,
-        expiryDateRangeFrom: dateRange.from?.getTime(),
-        expiryDateRangeTo: dateRange.to?.getTime(),
-        search: debouncedSearchFilter,
-        offset: pageNumber * PAGE_SIZE,
-        limit: PAGE_SIZE,
-        order: sortOrder,
+  const { refetch } = useQuery<GetPermitHoldersResponse, GetPermitHoldersRequest>(
+    GET_PERMIT_HOLDERS_QUERY,
+    {
+      variables: {
+        filter: {
+          userStatus: userStatusFilter,
+          permitStatus: permitStatusFilter,
+          expiryDateRangeFrom: dateRange.from?.getTime(),
+          expiryDateRangeTo: dateRange.to?.getTime(),
+          search: debouncedSearchFilter,
+          offset: pageNumber * PAGE_SIZE,
+          limit: PAGE_SIZE,
+          order: sortOrder,
+        },
       },
-    },
-    onCompleted: ({ applicants: { result, totalCount } }) => {
-      setPermitHolderData(
-        result.map(
-          ({
-            id,
-            firstName,
-            middleName,
-            lastName,
-            addressLine1,
-            addressLine2,
-            city,
-            postalCode,
-            ...applicant
-          }) => ({
-            id,
-            name: {
+      // ! Temporary fix to force permit holder row updates when navigating (skip cache)
+      fetchPolicy: 'network-only',
```

Comment:
how come this is only needed in the permit-holders table?

Reply:
Also will this refetch the query after every filter change? If so this might get a bit expensive and we should probably think of a more efficient long term solution imo

i found this after doing a bit of research that may be helpful https://www.apollographql.com/blog/apollo-client/caching/when-to-use-refetch-queries/

Reply:
Side note: could we use/implement something like “nextFetchPolicy” where only on page load we bypass cache and on subsequent queries we don’t?

https://www.apollographql.com/docs/react/data/queries/#nextfetchpolicy

Reply:
> how come this is only needed in the permit-holders table?
Not actually sure, I noticed that the permit holder table was not refetching as needed so I added this in as a temp fix.

As per our discussion today, we'll leave in this temporary fix for now in the interest of time, but I'll keep this comment  thread open and create a ticket to address this. Thanks for the catch!